### PR TITLE
feat: add banner output option

### PR DESCRIPTION
**Pull request by PengBoUESTC**
### Description
add `banner` option for config
**WIP**

### Test Plan

---

### CODE REVIEW

File: crates/rolldown/src/chunk/mod.rs [hunk 1]
```diff
@@ -155,7 +155,10 @@ impl Chunk {
           Ok(())
         },
       )?;
-
+    // add banner
+    if let Some(banner) = &output_options.banner {
+      concat_source.prepend_source(Box::new(RawSource::new(banner.to_string())))
+    }
```
Comment (underfin) [inline crates/rolldown/src/chunk/mod.rs:163]: Here `prepend_source` is unnecessary, you can move it to the first of chunk `concat_source` and using `addSource`.

Comment (PengBoUESTC) [reply crates/rolldown/src/chunk/mod.rs:163]: I'm not good at `rust` at all, so maybe U should review carefully +_+

**Review by underfin — APPROVED** [review]

**Comment by underfin** [comment]
Thank you.

File: crates/rolldown_binding/src/options/binding_output_options/mod.rs [hunk 2]
```diff
@@ -15,7 +18,10 @@ pub struct BindingOutputOptions {
 
   // amd: NormalizedAmdOptions;
   // assetFileNames: string | ((chunkInfo: PreRenderedAsset) => string);
-  // banner: () => string | Promise<string>;
+  #[derivative(Debug = "ignore")]
+  #[serde(skip_deserializing)]
+  #[napi(ts_type = "undefined | string | ((chunk: RenderedChunk) => string | Promise<String>)")]
```
Comment (hyf0) [inline crates/rolldown_binding/src/options/binding_output_options/mod.rs:23]: This should be
```
#[napi(ts_type = "string | ((chunk: RenderedChunk) => MaybePromise<VoidNullable<string>>")]
```

**Comment by PengBoUESTC** [comment]
> Thank you.

Thank U :);

File: crates/rolldown_binding/src/options/binding_output_options/mod.rs (hunk 2 above)
Comment (PengBoUESTC) [reply crates/rolldown_binding/src/options/binding_output_options/mod.rs:23]: Thank U :);

**Pull request merged**